scheduler/src/cook/mesos/api.clj

```diff
@@ -1556,138 +1555,159 @@
                                   :reasons (job-reasons conn job)})
                        (::jobs ctx)))}))
 
+(defn- streaming-json-encoder
+  "Takes as input the response body which can be converted into JSON,
+  and returns a function which takes a ServletResponse and writes the JSON
+  encoded response data. This is suitable for use with jet's server API."
+  [data]
+  (fn [^ServletResponse resp]
+    (cheshire/generate-stream data
+                              (OutputStreamWriter. (.getOutputStream resp)))))
+
+(defn- streaming-json-middleware
+  "Ring middleware which encodes JSON responses with streaming-json-encoder.
+  All other response types are unaffected."
+  [handler]
+  (fn streaming-json-handler [req]
+    (let [{:keys [headers body] :as resp} (handler req)
+          json-response (= "application/json" (get headers "Content-Type" "application/json"))]
```

Comment:
So I added this because the Swagger UI was broken when serving JSON to it's frontend. I think the way it was working before was that it was automatically converting maps and lists to JSON. Would you prefer that?